internal/pkg/earpointer/analysis_test.go

```diff
@@ -65,9 +77,15 @@ func TestFieldAddr(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	want := "{*f.a}: [x->f.t0], {f.a}: --> *f.a, {f.b}: [], {f.t0}: --> f.b"
-	if got := state.String(); got != want {
-		t.Errorf("got: %s\n want: %s", got, want)
+	want := map[string]string{
+		"{f.a}":  "--> *f.a",
+		"{*f.a}": "[x->f.t0]",
+		"{f.t0}": "--> f.b",
+		"{f.b}":  "[]",
+	}
+	wantstr := partitionItemsToString(want)
+	if got := state.String(); got != wantstr {
```

Comment:
Consider using `cmp.Diff` on the strings for a human-readable diff. (We have a few example uses in this codebase if you want a quick reference.)